### PR TITLE
Some more C# formatting and style fixes

### PR DESCRIPTION
**Pull request by aaronfranke**
Follow-up to #50757. This PR manually picks out some of the changes of #49352, anything that is easy to decide if it's a benefit.

This PR does the following:
* Use `_` for private fields/constants.
* Replace `extern static` with `static extern`.
* Replace `String` with `string`.
* Add explicit access modifiers.
* Add explicit parenthesis.
* Use camelCase for local variables.
* Use static types where 49352 does.
* Manually break up some of the lines that were automatically broken up by the formatter in 49352.

I can make a version of this PR for 3.x, if so #49352 will have to be rebased on top of it. As with #50757 the goal here is to break down #49352 into smaller pieces so we can more easily review the remaining style changes. There are also a few places where the automatic formatter produced sub-par results that I've replaced with better alternatives in this PR.

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/AABB.cs [hunk 1]
```diff
@@ -157,7 +157,10 @@ public Vector3 GetEndpoint(int idx)
                 case 7:
                     return new Vector3(_position.x + _size.x, _position.y + _size.y, _position.z + _size.z);
                 default:
-                    throw new ArgumentOutOfRangeException(nameof(idx), String.Format("Index is {0}, but a value from 0 to 7 is expected.", idx));
+                {
+                    string text = String.Format("Index is {0}, but a value from 0 to 7 is expected.", idx);
```
Comment (raulsntos) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/AABB.cs:161]: Should we use interpolated strings instead?
```suggestion
                    string text = $"Index is {idx}, but a value from 0 to 7 is expected.";
```

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Color.cs [hunk 2]
```diff
@@ -627,7 +627,9 @@ private static Color FromHTML(string rgba)
             }
             else
             {
-                throw new ArgumentOutOfRangeException("Invalid color code. Length is " + rgba.Length + " but a length of 6 or 8 is expected: " + rgba);
+                string text = "Invalid color code. Length is " + rgba.Length;
```
Comment (raulsntos) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Color.cs:630]: Should we use interpolated strings instead?
```suggestion
                string text =  $"Invalid color code. Length is {rgba.Length}, but a length of 6 or 8 is expected: {rgba}";
```

File: modules/mono/glue/GodotSharp/GodotSharp/Core/SignalAwaiter.cs [hunk 3]
```diff
@@ -42,12 +42,12 @@ public IAwaiter<object[]> GetAwaiter()
 
         internal void SignalCallback(object[] args)
         {
-            completed = true;
-            result = args;
+            _completed = true;
+            _result = args;
 
-            if (action != null)
+            if (_action != null)
```
Comment (raulsntos) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/SignalAwaiter.cs:48]: Delegate invocation can be simplified.
```csharp
action?.Invoke();
```

**Review by raulsntos — COMMENTED** [review]
LGTM, just a few nit-picks where I don't know what's preferred.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/AABB.cs [hunk 4]
```diff
@@ -157,7 +157,10 @@ public Vector3 GetEndpoint(int idx)
                 case 7:
                     return new Vector3(_position.x + _size.x, _position.y + _size.y, _position.z + _size.z);
                 default:
-                    throw new ArgumentOutOfRangeException(nameof(idx), String.Format("Index is {0}, but a value from 0 to 7 is expected.", idx));
+                {
+                    string text = $"Index is {idx}, but a value from 0 to 7 is expected.";
+                    throw new ArgumentOutOfRangeException(nameof(idx), text);
```
Comment (neikeq) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/AABB.cs:162]: Why a new variable rather than a line break?

File: modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs [hunk 5]
```diff
@@ -106,7 +106,11 @@ private static bool IsAnyOS(IEnumerable<string> names)
 
         public static string PathWhich([NotNull] string name)
         {
-            return IsWindows ? PathWhichWindows(name) : PathWhichUnix(name);
+            if (IsWindows)
+            {
+                return PathWhichWindows(name);
+            }
+            return PathWhichUnix(name);
```
Comment (neikeq) [inline modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs:113]: Why this change? Previous code looked more readable.

File: modules/mono/editor/GodotTools/GodotTools/Ides/Rider/RiderPathManager.cs [hunk 6]
```diff
@@ -49,15 +49,16 @@ public static void Initialize()
...
                 Globals.EditorDef(EditorPathSettingName, newPath);
                 editorSettings.SetSetting(EditorPathSettingName, newPath);
             }
         }
 
         public static bool IsExternalEditorSetToRider(EditorSettings editorSettings)
         {
-            return editorSettings.HasSetting(EditorPathSettingName) && IsRider((string)editorSettings.GetSetting(EditorPathSettingName));
+            bool hasSetting = editorSettings.HasSetting(EditorPathSettingName);
+            return hasSetting && IsRider((string)editorSettings.GetSetting(EditorPathSettingName));
```
Comment (neikeq) [inline modules/mono/editor/GodotTools/GodotTools/Ides/Rider/RiderPathManager.cs:61]: Why a new variable rather than a line break?

File: modules/mono/editor/GodotTools/GodotTools/Build/MsBuildFinder.cs [hunk 7]
```diff
@@ -165,7 +165,15 @@ private static string FindMsBuildToolsPathOnWindows()
...
-            var envNames = Internal.GodotIs32Bits() ? new[] {"ProgramFiles", "ProgramW6432"} : new[] {"ProgramFiles(x86)", "ProgramFiles"};
+            string[] envNames;
+            if (Internal.GodotIs32Bits())
+            {
+                envNames = new string[] { "ProgramFiles", "ProgramW6432" };
+            }
+            else
+            {
+                envNames = new[] { "ProgramFiles(x86)", "ProgramFiles" };
+            }
```
Comment (neikeq) [inline modules/mono/editor/GodotTools/GodotTools/Build/MsBuildFinder.cs:176]: Would look much better with inline if and line breaks after `?` and `:`.

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Vector3.cs [hunk 8]
```diff
@@ -371,10 +373,14 @@ public Axis MinAxis()
...
-            var vd = to - v;
-            var len = vd.Length();
-            return len <= delta || len < Mathf.Epsilon ? to : v + vd / len * delta;
+            Vector3 v = this;
+            Vector3 vd = to - v;
+            real_t len = vd.Length();
+            if (len <= delta || len < Mathf.Epsilon)
+            {
+                return to;
+            }
```
Comment (neikeq) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Vector3.cs:382]: No braces here

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs [hunk 9]
```diff
@@ -373,10 +376,14 @@ public Axis MinAxis()
...
-            var vd = to - v;
-            var len = vd.Length();
-            return len <= delta || len < Mathf.Epsilon ? to : v + vd / len * delta;
+            Vector2 v = this;
+            Vector2 vd = to - v;
+            real_t len = vd.Length();
+            if (len <= delta || len < Mathf.Epsilon)
+            {
+                return to;
+            }
```
Comment (neikeq) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs:385]: No braces here

File: modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs [hunk 10]
```diff
@@ -671,23 +671,26 @@ public static bool IsValidIdentifier(this string instance)
...
+            {
+                bool validChar = instance[i] == '_' ||
+                    (instance[i] >= 'a' && instance[i] <= 'z') ||
+                    (instance[i] >= 'A' && instance[i] <= 'Z') ||
+                    (instance[i] >= '0' && instance[i] <= '9');
 
                 if (!validChar)
+                {
                     return false;
+                }
```
Comment (neikeq) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs:693]: No braces here

File: modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs [hunk 11]
```diff
@@ -671,23 +671,26 @@ public static bool IsValidIdentifier(this string instance)
             int len = instance.Length;
 
             if (len == 0)
+            {
                 return false;
+            }
```
Comment (neikeq) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs:676]: No braces here

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs [hunk 12]
```diff
@@ -643,7 +648,11 @@ public static real_t Tanh(real_t s)
         public static int Wrap(int value, int min, int max)
         {
             int range = max - min;
-            return range == 0 ? min : min + ((value - min) % range + range) % range;
+            if (range == 0)
+            {
+                return min;
+            }
```
Comment (neikeq) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs:654]: No braces here

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs [hunk 13]
```diff
@@ -402,7 +402,11 @@ public static real_t Min(real_t a, real_t b)
         /// <returns>The value after moving.</returns>
         public static real_t MoveToward(real_t from, real_t to, real_t delta)
         {
-            return Abs(to - from) <= delta ? to : from + Sign(to - from) * delta;
+            if (Abs(to - from) <= delta)
+            {
+                return to;
+            }
```
Comment (neikeq) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Mathf.cs:408]: No braces here

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Color.cs [hunk 14]
```diff
@@ -800,7 +802,14 @@ public void ToHSV(out float hue, out float saturation, out float value)
...
 
-            saturation = max == 0 ? 0 : 1f - 1f * min / max;
+            if (max == 0)
+            {
+                saturation = 0;
+            }
+            else
+            {
+                saturation = 1 - (min / max);
+            }
```
Comment (neikeq) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Color.cs:812]: No braces here

**Review by neikeq — CHANGES_REQUESTED** [review]
I've highlighted the changes that didn't convince me.

Also our style is not to surround single line blocks with braces, unless there's another branch which isn't single line. For example, braces is fine here:

```
if (foo)
{
    WriteLine();
}
else
{
    Bar();
    foo = false;
}
```

Fixing existing code that doesn't follow this rule is not required, but the PR must not introduce new ones.

File: modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs (hunk 5 above)
Comment (aaronfranke) [reply modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs:113]: Do we want to only have Windows vs Unix, or do we want to make this code flexible in case we add more options? The if statement semantically says "check if this is Windows, if so use Windows, and have Unix as a fallback", while the ternary says to me "check if this is Windows, if so use Windows, else this is Unix".

File: modules/mono/editor/GodotTools/GodotTools/Ides/Rider/RiderPathManager.cs (hunk 6 above)
Comment (aaronfranke) [reply modules/mono/editor/GodotTools/GodotTools/Ides/Rider/RiderPathManager.cs:61]: Either way works, the line was too long as-is.

File: modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs (hunk 5 above)
Comment (neikeq) [reply modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs:112]: I wouldn't really worry about that too much. Supposing this ever needs to change, which is already unlikely, the change is very simple.
That being said, now that the braces were removed this is not that important.

**Review by neikeq — APPROVED** [review]

**Comment by neikeq** [comment]
Thanks!
